### PR TITLE
Rectify errors in UG 

### PR DESCRIPTION
**Pull request by nur-haziq**
### Refactoring:
- Refactored the flags description portion into a tabular format
&nbsp;
- Refactored summary table, and included details for `restock` and `sell` commands. (viewing the source diff for this is quite cancer, i think can just view the rich diff)
- #228 
&nbsp;
- Condense the command information for `list`
- #223 
- #205 

### Typo Corrections:
- Corrected the typo for the `sell` by index command 
- #206 
- #199 
&nbsp;
- Corrected the typo for the `restock` by index command
- #230 
- #192 
- #212 

### Redundancy Elimination:
- Remove repetitive examples from the documentation 
- #227 

---

Please review the changes and let me know if there's anything else that I have missed out on. Thanks :)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -374,77 +383,44 @@ Format: `search -n NAME_QUERY -d DESCRIPTION_QUERY -q QUANTITY_RANGE -c COST_PRI
...
-
-Format: `list -e`
-
-Example:
-- Item A, Expiry Date: 2023-05-01
-- Item B, Expiry Date: 2023-04-15
-- Item C, Expiry Date: 2023-06-20
-
-After using `list -e`, the sorted list will be:
+If we were to use `list -e` however, the sorted list by expiry date will then be:
```
Comment (YHWong20) [inline docs/UserGuide.md:417]: Do you reckon that it'll be necessary for us to mention that items w/o expiry dates will not be displayed?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -374,77 +383,44 @@ Format: `search -n NAME_QUERY -d DESCRIPTION_QUERY -q QUANTITY_RANGE -c COST_PRI
...
 - Item B, Expiry Date: 2023-04-15
 - Item A, Expiry Date: 2023-05-01
 - Item C, Expiry Date: 2023-06-20
-- 
-#### List Inventory (Sorted Based on Item Profit)
-
-Used to view the inventory based on the profit of items, in descending order:
 
-Format: `list -p`
+Similarly, you can use `list -s` and `list -p`, to sort by sale price and profit respectively.
```
Comment (YHWong20) [inline docs/UserGuide.md:422]: Likewise, operational items w/o sale prices will not be displayed here.

**Review by YHWong20 — APPROVED** [review]
Thanks for addressing these issues!!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -374,77 +383,44 @@ Format: `search -n NAME_QUERY -d DESCRIPTION_QUERY -q QUANTITY_RANGE -c COST_PRI
...
-Used to view the inventory in the order items were added:
+View the inventory in the order items were added:
 
 Format: `list`
 
-#### List inventory (sorted based on item cost price)
+#### List Inventory (Sorted)
 
-Used to view the inventory based on the sale price of items.
+BinBash also allows you to sort your inventory based on different criteria using the following flags:
```
Comment (nkotaa) [inline docs/UserGuide.md:394]: A brief clarification that this displays results in ascending order would be useful.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -674,15 +645,17 @@ Similarly, your saved data will be automatically loaded into BinBash when you st
...
+| **update**  | `update -n ITEM_NAME -d ITEM_DESCRIPTION -q ITEM_QUANTITY -e EXPIRY_DATE -s SALE_PRICE -c COST_PRICE -t THRESHOLD` <br> `update -i ITEM_INDEX -d ITEM_DESCRIPTION -q ITEM_QUANTITY -e EXPIRY_DATE -s SALE_PRICE -c COST_PRICE -t THRESHOLD` | Updates the details of an existing item in the inventory.                                                        |
+| **profit**  | `profit`                                                                                                                                                                                                                                    | Displays the total profit earned from the inventory.                                                             |
+| **bye**     | `bye`                                                                                                                                                                                                                                       | Exits the application.                                                                                           |
```
Comment (nkotaa) [inline docs/UserGuide.md:658]:
```suggestion
| **bye**     | `bye`  <br> `exit` <br> `quit`                                                                                                                                                                                                                                     | Exits the application.                                                                                           |
```
all work as well.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -674,15 +645,17 @@ Similarly, your saved data will be automatically loaded into BinBash when you st
...
+| **search**  | `search -n NAME_QUERY -d DESCRIPTION_QUERY -q QUANTITY_RANGE -c COST_PRICE_RANGE -s SALE_PRICE_RANGE -e EXPIRY_DATE_RANGE -l NUMBER_OF_RESULTS`                                                                                             | Searches for items in the inventory based on various criteria.                                                   |
+| **list**    | `list` <br> `list -c` <br> `list -s` <br> `list -e` <br> `list -p`                                                                                                                                                                          | Lists all items in the inventory, with optional sorting based on cost price, sale price, expiry date or profits. |
+| **delete**  | `delete -i ITEM_INDEX` <br> `delete -n **ITEM_NAME**`                                                                                                                                                                                       | Deletes an item from the inventory.                                                                              |
```
Comment (nkotaa) [inline docs/UserGuide.md:653]:
```suggestion
| **delete**  | `delete -i ITEM_INDEX` <br> `delete -n ITEM_NAME`                                                                                                                                                                                       | Deletes an item from the inventory.                                                                              |
```
Possible formatting error.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -674,15 +645,17 @@ Similarly, your saved data will be automatically loaded into BinBash when you st
...
+| **Command** | **Usage**                                                                                                                                                                                                                                   | **Description**                                                                                                  |
+|-------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
+| **add**     | `add -re -n ITEM_NAME -d ITEM_DESCRIPTION -q ITEM_QUANTITY -s SALE_PRICE -c COST_PRICE -t THRESHOLD` <br> `add -op -n ITEM_NAME -d ITEM_DESCRIPTION -q ITEM_QUANTITY -e EXPIRY_DATE -s SALE_PRICE -t THRESHOLD`                             | Adds a new item to the inventory.                                                                                |
```
Comment (nkotaa) [inline docs/UserGuide.md:650]:
```suggestion
| **add**     | `add -re -n ITEM_NAME -d ITEM_DESCRIPTION -q ITEM_QUANTITY -e EXPIRY_DATE -s SALE_PRICE -c COST_PRICE -t THRESHOLD` <br> `add -op -n ITEM_NAME -d ITEM_DESCRIPTION -q ITEM_QUANTITY -e EXPIRY_DATE -c COST_PRICE -t THRESHOLD`                             | Adds a new item to the inventory.                                                                                |
```

**Review by nkotaa — COMMENTED** [review]
Kindly mend inconsistencies.

**Pull request merged**